modules/core/src/main/java/org/apache/synapse/rest/Resource.java

```diff
@@ -310,10 +311,17 @@ void process(MessageContext synCtx) {
                             statisticReportingIndex, true);
                 }
                 return;
+            } else {
+                // TODO Need to finalize
```

Comment:
what is remaining here to be finalized?

Reply:
Had to finalize how to handle cors headers when respond mediator is used.